app/src/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -1837,11 +1838,13 @@ public void handleFormLoadCompletion(FormController fc) {
      * Call when the user provides input that they want to quit the form
      */
     private void triggerUserQuitInput() {
-        //If we're just reviewing a read only form, don't worry about saving
-        //or what not, just quit
-        if(mFormController.isFormReadOnly()) {
-            //It's possible we just want to "finish" here, but
-            //I don't really wanna break any c compatibility
+        if(!formHasLoaded()) {
+            finish();
```

Comment:
Yeah it is mostly unrelated; trying to make form entry erroring more robust. I found that if you dismissed the form load error dialog (which shouldn't be possible possible anymore) on form entry without finishing back to the previous activity, CommCare would crash when you tried to leave Form Entry because it assumed there was a form when running the exit code.